### ISSUE TITLE
Covariance Estimation

### ISSUE THREAD

**Issue by zmarkovich** [grf-labs/grf#1029]
Is it possible to estimate the covariance of predictions made using a causal or regression forest or are there any plans to implement such a feature? 

Such estimates would be very useful for analyzing the asymptotic distribution of several estimates using monte carlo methods, but don't seem to be straightforward in the current package?

**Comment by erikcs** [comment]
Hi @zmarkovich, could you please elaborate what you intend to do?

**Comment by zmarkovich** [comment]
Hi @erikcs. Thanks for the reply.

Basically, I'd like to be able to sample from the asymptotic distribution of the predictions from the random forest. Athey et al show the predictions are going to be asymptotically normal, and the predict method provides estimates of the mean and variance of those predictions, so all that's missing are the covariance estimates. So I was wondering if there's a way to obtain such covariance estimates. Does that make sense?

**Comment by swager** [comment]
@zmarkovich yes it would in principle be possible to extend the analysis and methods from the GRF paper to cover the sampling distribution of predictions at multiple points (including the covariance of predictions). However, this avenue of research is open for the time being (and we don't have the required functionality implemented in GRF).
